### ISSUE TITLE
Ignore ignored statements (restroom-mw) when checking for collisions 

### ISSUE THREAD

**Issue by andrea-dintino** [dyne/Zenroom#283]
This script should not return an error:

```
Rule unknown ignore
Scenario 'ecdh': Create the keypair
Given that I have an endpoint named 'endpoint1'
Given that I have an endpoint named 'endpoint2'
Given I have a 'string dictionary' named 'myData'
Given I connect to 'endpoint1' and save the output into 'myData'
Then print 'myData'
Then pass the output to 'endpoint2'
```

data: 

```
{
	"endpoint1": "http://showcase.api.linx.twenty57.net/UnixTime/tounix?date=now",
	"endpoint2": "https://dyneorg.free.beeceptor.com"
}
```

Now it returns: 

Release version: 2.0.0\n .  Build commit hash: b18ac4bc\n .  ECDH curve is SECP256K1\n .  ECP curve is BLS381\n .  Memory in use: 345 KB\n .  Rule unknown ignore\n[W] Zencode pattern ignored: Given that I have an endpoint named 'endpoint1'\t1\n[W] Zencode pattern ignored: Given that I have an endpoint named 'endpoint2'\t1\n[W] Zencode pattern ignored: Given I connect to 'endpoint1' and save the output into 'myData'\t1\n[W] Zencode pattern ignored: Then pass the output to 'endpoint2'\t1\n[W] Zencode is missing version check, please add: rule check version N.N.N\n[!] ERROR:\n[!] EXEC: /zencode.lua:592: Object name collision in input: endpoint2\n[!] Execution aborted\n[!] Error detected. Execution aborted.\n[*] Zenroom teardown.\n .  Memory used: 375 KB\n

**Comment by jaromil** [comment]
Next commit adds `rule collision ignore` to deactivate HEAP namespace collision check.

Also it automatically configures it if `rule caller restroom-mw` is used, implying both:
```
rule caller restroom-mw = 
  rule unknown ignore
  rule collision ignore
```

I recommend using `rule caller restroom-mw` from now on so I can add more specific features there without you needing to change examples and scripts again.

**Comment by jaromil** [comment]
fixed in 7ea446dd05d3ebd23cf028848c3160c74a3756b0
